### ISSUE TITLE
Add ability to register custom container components.

### ISSUE THREAD

**Issue by wujekbogdan** [pojome/elementor#6625]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

I'd like to build an extension for Elementor. My extension would require a wrapper component (like the Inner Section component) that would hold the settings for child components. 

For example, the root component would query a post (let's say WooCommerce product) and hold a reference to that post. Inner components would be like product_price component, product_image component, product_in_stock_count component,  etc - they would display the data within the context of the parent component.

I already tried to extend the `\Elementor\Element_Column` class and register it using the `elementor/elements/elements_registered` hook as follows:

``` php
// widget.php
class Test_Widget extends \Elementor\Element_Column {

    public function get_name() {
        return 'test widget';
    }

    public function get_title() {
        return __( 'test widget', 'plugin-name' );
    }
}

// my-plugin.php
add_action('elementor/elements/elements_registered', function () {
    require_once __DIR__ .DIRECTORY_SEPARATOR . 'widget.php';
    Elementor\Plugin::instance()->elements_manager->register_element_type( new \Test_Widget() );
});
```

But it doesn't seem to work. The `test widget` component does't show up in the UI. I dug into the code a little but I don't fully understand how the elements are registered. Probably I'm not doing it right. 

**Describe the solution you'd like**

Ideally, the container component would provide data to child components via props. So the child components would be able to get the data from their parent.

So, basically, it would work the same way as [props work in Vue library](https://vuejs.org/v2/guide/components-props.html) or [props in React](https://reactjs.org/docs/components-and-props.html).

So the `Widget_Base` would have a method, let's say `get_parent_context()` or `get_props()`, that would return the data providerd by the parent.

**Additional context**

IMO such a thing would be a killer feature. There's no page builder plugin on the market that has such a feature. It would give Elementor a very wide range of possibilities for future extensions.

---
If I'm wrong, and registering a custom container element is possible, but not documented, then I'll appreciate some tips on how to register it.

Thanks!

**Comment by bainternet** [comment]
Currently, this feature is not supported, but we will look into it in the future if we see a large demand for it.

I merged this issue with  #2587, which is on the same subject so we could deal with it more efficiently. 
Please feel free to continue the conversation there.

Thanks for your feedback!

**Comment by nicholaszein** [comment]
Hello @wujekbogdan!

It's been a while, but I wanted to let you know that we now have the Flexbox Container since **v3.6**, which is now a stable experiment since **v3.8**!

This solves this issue.

Thanks for being so patient! 🙏 

✔️ Feel free to update to the latest version of Elementor and enable the Container experiment.

Cheers 🥂
